src/components/widgets/Map.vue

```diff
@@ -502,58 +640,138 @@ watch(vehiclePositionHistory, (newPoints) => {
-      menuPosition.top = `${event.originalEvent.clientY - y}px`
-    }
-  } else {
-    console.error('Invalid event structure:', event)
+const gotoMarker = ref<L.Marker>()
+let didGoToMarker = false
+
+const setDefaultMapPosition = async (): Promise<void> => {
+  if (!map.value || !clickedLocation.value) return
+
+  try {
+    await missionStore.setDefaultMapPosition(clickedLocation.value, zoom.value)
+    openSnackbar({ message: 'Default map position set', variant: 'success' })
+
+    const tempMarker = L.marker(clickedLocation.value as LatLngTuple, {
+      icon: L.divIcon({
+        className: 'marker-icon',
+        iconSize: [24, 24],
+        iconAnchor: [12, 12],
+      }),
+    }).addTo(map.value)
+
+    const tempTooltip = L.tooltip({
+      content: '<i class="mdi mdi-map-check text-[18px] border-[1px] rounded-full px-[2px] py-[1px]"></i>',
+      permanent: true,
+      direction: 'center',
+      className: 'waypoint-tooltip',
+      opacity: 1,
+    })
+    tempMarker.bindTooltip(tempTooltip).openTooltip()
+
+    const markerEl = tempMarker.getElement() as HTMLElement | null
+    const tooltipEl = tempMarker.getTooltip()?.getElement() as HTMLElement | null
+
+    ;[markerEl, tooltipEl].forEach((el) => {
+      if (el) {
+        el.style.transition = 'opacity 1s'
+        el.style.opacity = '1'
+      }
+    })
+
+    setTimeout(() => {
+      ;[markerEl, tooltipEl].forEach((el) => {
+        if (el) el.style.opacity = '0'
+      })
+      setTimeout(() => {
+        map.value?.removeLayer(tempMarker)
+      }, 1000)
+    }, 1500)
+  } catch (error) {
+    console.error(error)
+    openSnackbar({ message: 'Failed to set default map position', variant: 'error' })
   }
 }
 
-const onMenuOptionSelect = async (option: string): Promise<void> => {
-  switch (option) {
-    case 'goto':
-      if (clickedLocation.value) {
-        // Define default values
-        const hold = 0
-        const acceptanceRadius = 0
-        const passRadius = 0
-        const yaw = 0
-        const altitude = vehicleStore.coordinates.altitude ?? 0
+const executeGoToOption = (): void => {
+  if (!clickedLocation.value || !map.value) return
 
-        const latitude = clickedLocation.value[0]
-        const longitude = clickedLocation.value[1]
+  if (gotoMarker.value) {
+    map.value.removeLayer(gotoMarker.value)
+    gotoMarker.value = undefined
+  }
 
+  gotoMarker.value = L.marker(clickedLocation.value as LatLngTuple, {
+    icon: L.divIcon({
+      className: 'marker-icon',
+      iconSize: [24, 24],
+      iconAnchor: [12, 12],
+    }),
+  }).addTo(map.value)
+
+  const gotoTooltip = L.tooltip({
+    content: '<i class="mdi mdi-crosshairs-gps border-[1px] rounded-full text-[18px] px-[2px] pt-[1px] "></i>',
+    permanent: true,
+    direction: 'center',
+    className: 'waypoint-tooltip',
+    opacity: 1,
+  })
+  gotoMarker.value.bindTooltip(gotoTooltip)
+
+  didGoToMarker = false
+  Promise.resolve(
+    slideToConfirm(
```

Comment:
This new GoTo logic that was added here is wrong. The `vehicleStore.goTo` already has a complete slide-to-confirm logic that prevents duplicated slider requests to appear (because of the need for an Arm request in sequence).